### PR TITLE
Change Docker layers to benefit more from caching layers

### PR DESCRIPTION
**Pull request by marcofranssen**
This reduces the amount of layers that have to be downloaded when new releases are done.

As well it reduces the amount of layers that have to be rebuild or pushed.

### CODE REVIEW

File: Dockerfile.frontend-container [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
+CMD ["npm", "start"]
 
 # set env variables
-ENV REACT_APP_API_SERVER_URI $REACT_APP_API_SERVER_URI
-ENV REACT_APP_AUTH_SERVER_URI $REACT_APP_AUTH_SERVER_URI
-ENV REACT_APP_SPIRE_HEALTH_CHECK_ENABLE $REACT_APP_SPIRE_HEALTH_CHECK_ENABLE
+ENV REACT_APP_API_SERVER_URI $REACT_APP_API_SERVER_URI \
+    REACT_APP_AUTH_SERVER_URI $REACT_APP_AUTH_SERVER_URI \
+    REACT_APP_SPIRE_HEALTH_CHECK_ENABLE $REACT_APP_SPIRE_HEALTH_CHECK_ENABLE \
+    PORT_FE=3000
```
Comment (marcofranssen) [inline Dockerfile.frontend-container:10]: changes from 4 independent layers to just a single layer.

File: Dockerfile.backend-container [hunk 2]
```diff
@@ -2,10 +2,10 @@ FROM alpine
...
 WORKDIR /opt/spire
+ENTRYPOINT ["/opt/spire/run_backend.sh"]
 
-COPY bin/tornjak-backend tornjak-backend
 COPY sample-keys sample-keys
-
 # Add init
 COPY run_backend.sh run_backend.sh
-ENTRYPOINT ["/opt/spire/run_backend.sh"]
+COPY bin/tornjak-backend tornjak-backend
```
Comment (marcofranssen) [inline Dockerfile.backend-container:10]: Moving this to the end makes the things most likely to change on new images in the last docker layer, meaning the layers before can all be reused as is. No pulls and pushes and rebuilds required for those layers.

File: Dockerfile.frontend-container [hunk 3]
```diff
@@ -1,19 +1,18 @@
...
 
-ENV PORT_FE=3000
 EXPOSE $PORT_FE
 
-CMD ["npm", "start"]
\ No newline at end of file
+COPY tornjak-frontend .
+
+RUN npm install --omit=dev \
```
Comment (marcofranssen) [inline Dockerfile.frontend-container:16]: Same here, doing this last makes it possible to reuse the other layers as is. no more pulling/pushing required of those.

File: Dockerfile.frontend-container (hunk 1 above)
Comment (mamy-CS) [reply Dockerfile.frontend-container:10]: Thank you for this layer improvement!

File: Dockerfile.backend-container (hunk 2 above)
Comment (mamy-CS) [reply Dockerfile.backend-container:10]: Thank you, makes sense!

**Comment by mamy-CS** [comment]
Thank you @marcofranssen for these improvements. Even though we gained only a couple of Mbs. Once we move completely to production, it should be optimized to the max.

**Review by mamy-CS — APPROVED** [review]
lgtm!

**Pull request merged**